### PR TITLE
Update types_of_testing.md

### PR DESCRIPTION
**Pull request by DivineDominion**
I corrected a few typos and re-formulated a few sentences to better match how I understand them. This could be wrong, of course :)

I also expanded the topic overviews a bit to frame the following explanations.

---

@orta I'd love to include a few words on the difference between Unit Testing and Functional Testing/Integration Tests, but I'm not sure if understanding the difference is important for a _pragmatic_ book.

### CODE REVIEW

File: chapters/en-UK/xctest/types_of_testing.md [hunk 1]
```diff
@@ -1,21 +1,23 @@
 # Types of Testing
 
-A test is a way of verifying code, and making your assumptions explicit. When you make a change somewhere, tests exist to show the implicit connections between objects.
```
Comment (DivineDominion) [inline chapters/en-UK/xctest/types_of_testing.md]: How do tests reveal implicit connections? Do you mean they'll fail when you change some algorithm somewhere?

File: chapters/en-UK/xctest/types_of_testing.md [hunk 2]
```diff
@@ -1,21 +1,23 @@
...
+In Cocoa, this is less of a problem as we can rely on the tooling more. Given a compiler, static type systems, and a well integrated static analyser -- you can know if your code is going to break someone else's very quickly. However, this is only a lint, the code compiling doesn't mean that it will still correctly show a view controller when you tap a button. In other words, the behaviour of your app can still be wrong.
 
-That is what testing is for. I will cover three major topics in this book - there are many, many other types of testing patterns ( for example, last week I was introduced to Model Testing. ) However, These are the dominent patterns that exist within the Cocoa world, and I have experience with them. Otherwise, I'd just be copy & pasting from Wikipedia.
+That is what testing is for. I will cover three major topics in this book. There are many, many other types of testing patterns -- for example, last week I was introduced to Model Testing. However, these three topics are the dominent patterns that exist within the Cocoa world, and I have experience with them. Otherwise, I'd just be copy & pasting from Wikipedia.
```
Comment (DivineDominion) [inline chapters/en-UK/xctest/types_of_testing.md]: "Patterns" is misleading; if I had to, I'd call mocking a testing pattern. 

Also, BDD and Unit Testing don't really belong to the same category of things (although both headings end in "Testing" :)). Here's a proposal:

1. Granularity: Unit Tests, Functional/Integration Tests
2. Styles: TDD with the xUnit convention (starting test cases with "test") vs BDD with the well-known style you already mentioned

Comment (orta) [inline chapters/en-UK/xctest/types_of_testing.md]: Personally, I think the xUnit style is too barebones to be useful when put next BDD style tests. I see it as a subset of BDD tests that you should know exists then move on to a better abstraction for writing tests ASAP. 

E.g. for me, this was enough to show that existed, https://github.com/orta/pragmatic-testing/blob/master/chapters/en-UK/xctest/what_is_xctest_how_does_it_work.md

File: chapters/en-UK/xctest/types_of_testing.md (hunk 1 above)
Comment (orta) [inline chapters/en-UK/xctest/types_of_testing.md]: This tweet kind of sums up that idea for me - https://twitter.com/CodeWisdom/status/747469307022544897

There's a point where you're not aware of the full context for any change, tests help make those connections more implicit

**Comment by orta** [comment]
WRT:

>  I'd love to include a few words on the difference between Unit Testing and Functional Testing/Integration Tests, but I'm not sure if understanding the difference is important for a pragmatic book.

Pragmatically, it's XCTest vs [UITesting](https://developer.apple.com/library/ios/documentation/DeveloperTools/Conceptual/InstrumentsUserGuide/UIAutomation.html) 

I covered this a bit in [Integration Testing](https://github.com/orta/pragmatic-testing/blob/master/chapters/en-UK/xctest/integration_testing.md) - but am definitely open to adding more around the mental model differences 👍

**Comment by orta** [comment]
And now that I've read the diff - I think this is a great improvement! Thanks @DivineDominion

File: chapters/en-UK/xctest/types_of_testing.md (hunk 2 above)
Comment (DivineDominion) [inline chapters/en-UK/xctest/types_of_testing.md]: "Usefulness" seems to be not a good fit as you can transform BDD-style tests into xUnit tests. They're more readable and all, I admit. But there's no _need_ to use them in order to test an app properly. Will keep this in mind, though.

Comment (orta) [inline chapters/en-UK/xctest/types_of_testing.md]: I think that's BDD->xUnit a lossy transition though, whereas xUnit->BDD isn't. The main thing that I care about here is the `describe` and  `before`/`after` structure that you get that makes it significantly easier to write smaller, more focused tests. 

You can obviously refactor to make all your own functions to work around that but the xUnit flatness makes mean that your code is now totally intermingled with helper functions and `testXXX` functions.

**Pull request merged**